### PR TITLE
Add GHA to push gotrue image to Quay.io

### PR DESCRIPTION
**Pull request by terbepetra**
## Describe your changes
Added a new Github action to be able to push image to our Quay.io repository.

## How best to test these changes
The GHA will run with the pull_request, but the image will be only pushed when the pr is merged.

### CODE REVIEW

**Review by rbarabas — APPROVED** [review]

**Review by rbarabas — DISMISSED** [review]
Deprecated

File: .github/workflows/build-push-gotrue.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Get short github SHA
+        id: var
+        shell: bash
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
```
Comment (rbarabas) [inline .github/workflows/build-push-gotrue.yml:41]: This syntax is deprecated.

Comment (rbarabas) [reply .github/workflows/build-push-gotrue.yml:41]: Please review https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Comment (rbarabas) [reply .github/workflows/build-push-gotrue.yml:41]: We should do a cursory sweep to see if `set-output` is being used widely and fix it where possible. The switch to the new syntax is not bad. You can check in the build workflows I submitted recently to `go/terraform-modules` and `go/infra`.

Comment (terbepetra) [reply .github/workflows/build-push-gotrue.yml]: Thank you. Changed to env file

**Review by rbarabas — APPROVED** [review]
LGTM!

**Pull request merged**